testing/tests/test_context_on.py

```diff
@@ -344,3 +347,333 @@ def test_relation_departed_event():
+    icmpport: ops.Port
+    storage: ops.Storage
+
+    def __init__(self, handle: ops.Handle, **kwargs: typing.Any):
+        super().__init__(handle)
+        for key, value in kwargs.items():
+            setattr(self, key, value)
+
+    def snapshot(self):
+        base = super().snapshot()
+        # This loses a lot of the details, but for the test all we care about is
+        # that the type and the 'primary key' are correct.
+        base["cloudcredential"] = self.cloudcredential.auth_type
+        base["cloudspec"] = self.cloudspec.name
+        base["secret"] = self.secret.id
+        base["relation"] = self.relation.name
+        base["peerrelation"] = self.peerrelation.name
+        base["subordinaterelation"] = self.subordinaterelation.name
+        base["relation_id"] = self.relation.id
+        base["peerrelation_id"] = self.peerrelation.id
+        base["subordinaterelation_id"] = self.subordinaterelation.id
+        base["notice_id"] = self.notice.id
+        base["notice_key"] = self.notice.key
+        base["checkinfo"] = self.checkinfo.name
+        base["container"] = self.container.name
+        base["errorstatus"] = self.errorstatus.message
+        base["activestatus"] = self.activestatus.message
+        base["blockedstatus"] = self.blockedstatus.message
+        base["maintenancestatus"] = self.maintenancestatus.message
+        base["waitingstatus"] = self.waitingstatus.message
+        base["tcpport"] = self.tcpport.port
+        base["udpport"] = self.udpport.port
+        base["storage_name"] = self.storage.name
+        base["storage_index"] = self.storage.index
+        return base
+
+    def restore(self, snapshot: dict[str, typing.Any]):
+        super().restore(snapshot)
+        self.cloudcredential = ops.CloudCredential(
+            auth_type=snapshot["cloudcredential"]
+        )
+        self.cloudspec = ops.CloudSpec("", snapshot["cloudspec"])
+        self.secret = self.framework.model.get_secret(id=snapshot["secret"])
+        relation = self.framework.model.get_relation(
+            snapshot["relation"], relation_id=snapshot["relation_id"]
+        )
+        assert relation is not None
+        self.relation = relation
+        peerrelation = self.framework.model.get_relation(
+            snapshot["peerrelation"], relation_id=snapshot["peerrelation_id"]
+        )
+        assert peerrelation is not None
+        self.peerrelation = peerrelation
+        subordinaterelation = self.framework.model.get_relation(
+            snapshot["subordinaterelation"],
+            relation_id=snapshot["subordinaterelation_id"],
+        )
+        assert subordinaterelation is not None
+        self.subordinaterelation = subordinaterelation
+        self.notice = ops.pebble.Notice(
+            snapshot["notice_id"],
+            None,
+            "",
+            snapshot["notice_key"],
+            datetime.datetime.now(),
+            datetime.datetime.now(),
+            datetime.datetime.now(),
```

Comment:
Minor: is the value significant?
I wonder if testing with a fixed value would be cleaner than 3 teenily-tinily different current values.

Reply:
Nothing in the tests use any of the times, and the three timestamps could be different in practice, but I agree it's tidier for this test to use one value.